File: Algorithms/Python3.x/942-DI_String_Match.py
# Runtime: 80 ms, faster than 44.01% of Python3 online submissions for DI String Match.
# Memory Usage: 15 MB, less than 5.00% of Python3 online submissions for DI String Match.
class Solution:
    def diStringMatch(self, S: str) -> List[int]:
        A = []
        low, high = 0, len(S)
        for s in S:
            if s == 'I':
                A.append(low)
                low += 1
            else:
                A.append(high)
                high -= 1
        return A + [high]

File: Algorithms/C++/942-DI_String_Match.cpp
// Runtime: 44 ms, faster than 17.80% of C++ online submissions for DI String Match.
// Memory Usage: 10.2 MB, less than 42.86% of C++ online submissions for DI String Match.
class Solution {
public:
    vector<int> diStringMatch(string S) {
        vector<int> A;
        int low = 0, high = S.size();
        for (char s : S)
            if (s == 'I')
                A.push_back(low++);
            else
                A.push_back(high--);
        A.push_back(high);
        return A;
    }
};
